test/LegacyConfigTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-di for the canonical source repository
+ * @copyright Copyright (c) 2005-2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-di/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Di;
+
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\Error\Deprecated as DeprecatedError;
+use Zend\Di\LegacyConfig;
+use Zend\Di\Exception;
+
+/**
+ * @coversDefaultClass Zend\Di\LegacyConfig
+ */
+class LegacyConfigTest extends TestCase
+{
+    public function provideMigrationConfigFixtures()
+    {
+        $iterator = new \GlobIterator(__DIR__ . '/_files/legacy-configs/*.php');
+
+        /** @var \SplFileInfo $file */
+        foreach ($iterator as $file) {
+            $data = include $file->getPathname();
+            yield [
```

Comment:
Add names for each case. (You can do this with yield (`yield $name => $value`).)  In this case, I'd use the `$data['config']` value for the key.

Reply:
`yield $name => $value` does not work. PHPUnit  uses iterator_to_array without preserving keys: https://github.com/sebastianbergmann/phpunit/blob/master/src/Util/Test.php#L527
I'll change it to building an old fashioned array.

Reply:
It _does_ work; I used this technique just yesterday, and had it reporting perfectly. :smile: Give it a try!